tests/capture/transforms/test_capture_single_qubit_fusion.py

```diff
@@ -0,0 +1,699 @@
+        assert qfunc_jaxpr.eqns[3].primitive == qml.Hadamard._primitive
+        assert qfunc_jaxpr.eqns[4].primitive == qml.PauliX._primitive
+        assert qfunc_jaxpr.eqns[5].primitive == qml.PauliX._primitive
+        assert qfunc_jaxpr.eqns[6].primitive == qml.ops.Prod._primitive
+
+        result = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.literals, input)[0]
+        assert qml.math.allclose(result, np.zeros((3, 3)))
+
+    def test_transform_higher_order_primitive(self):
+        """Test that the inner_jaxpr of transform primitives is not transformed."""
+
+        @qml.transform
+        def fictitious_transform(tape):
+            """Fictitious transform"""
+            return [tape], lambda res: res[0]
+
+        @SingleQubitFusionInterpreter()
+        def circuit():
+            @fictitious_transform
+            def g():
+                qml.S(0)
+                qml.T(0)
+
+            qml.RX(0.1, 0)
+            g()
+            qml.RY(0.2, 0)
+
+        jaxpr = jax.make_jaxpr(circuit)()
+
+        assert len(jaxpr.eqns) == 3
+
+        assert jaxpr.eqns[0].primitive == qml.RX._primitive
+        assert jaxpr.eqns[1].primitive == fictitious_transform._primitive
+        assert jaxpr.eqns[2].primitive == qml.RY._primitive
+
+        inner_jaxpr = jaxpr.eqns[1].params["inner_jaxpr"]
+        assert len(inner_jaxpr.eqns) == 2
+        assert inner_jaxpr.eqns[0].primitive == qml.S._primitive
+        assert inner_jaxpr.eqns[1].primitive == qml.T._primitive
+
+    @pytest.mark.parametrize(
+        "selector, expected_ops",
+        [
+            (
+                0.2,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Hadamard(wires=[0]),  # H(0) and H(1) are not fused
+                    qml.Hadamard(wires=[1]),  # because they act on different wires
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.6,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Rot(np.pi / 2, 0.6, 0.0, wires=[0]),  # RX and S fused
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.8,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Rot(np.pi / 2, 0.8, 0.0, wires=[0]),  # RX and S fused
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+        ],
+    )
+    def test_cond(self, selector, expected_ops):
+        """Test that operations inside a conditional block are correctly fused."""
+
+        @SingleQubitFusionInterpreter()
+        def circuit(x):
+
+            qml.CNOT(wires=[0, 1])
+
+            def true_branch(x):
+                qml.RX(x, wires=0)
+                qml.S(wires=0)
+
+            # pylint: disable=unused-argument
+            def false_branch(x):
+                qml.H(0)
+                qml.H(1)
+
```

Comment:
Maybe for locality :) 
```suggestion
            def true_branch(x):
                qml.RX(x, wires=0)
                qml.S(wires=0)

            # pylint: disable=unused-argument
            def false_branch(x):
                qml.H(0)
                qml.H(1)

            qml.CNOT(wires=[0, 1])
```